backend/ccm/background_tasks/enroll_um_users_task.py

```diff
@@ -113,5 +113,61 @@ def handle_enrollment_results(enrollment_params, results, request, uniqname, req
         # This might happen when new scopes are added after the token was issued, but not going to be an issue with Prod release 
         logger.warning(f"Deleting CanvasOAuth2Token for user {uniqname} due to insufficient scopes on access token.")
         CanvasOAuth2Token.objects.filter(user=request.user).delete()
+    
+    email_enrollment_summary(
+        req_user_email=req_user_email,
+        course_id=course_id,
+        failed_enrollments=failed_enrollments,
+        enrollment_count=len(enrollment_params)
+    )
+
+def email_enrollment_summary(req_user_email, course_id, failed_enrollments, enrollment_count):
+    """
+    Compose and send enrollment result email, with CSV attachment if there are failures.
+    """
+    total = enrollment_count
+    failed = len(failed_enrollments)
+    succeeded = total - failed
+    course_canvas_link = f'https://{settings.CANVAS_OAUTH_CANVAS_DOMAIN}/courses/{course_id}'
+
+    email_subject = f"For course {course_id}, {succeeded}/{total} enrollments finished successfully" + (f" ({failed} failed)" if failed > 0 else "")
 
-    logger.info(email_subject)
\ No newline at end of file
+    # Use HTML for the body, with course_canvas_link as a hyperlink
+    success_body = (
+        f"For Course <a href='{course_canvas_link}'>{course_id}</a> enrolling all users is success"
+    )
+    failure_body = (
+        f"For Course <a href='{course_canvas_link}'>{course_id}</a> enrolling users encountered failures. See attachment for error list."
+    )
+    body = success_body if succeeded == total else failure_body
+
+    attachment = None
+    if failed_enrollments:
+        try:
+            output = io.StringIO()
+            fieldnames = ['sectionId', 'LoginId', 'role', 'ReasonForFailure']
+            writer = csv.DictWriter(output, fieldnames=fieldnames)
+            writer.writeheader()
+            for item in failed_enrollments:
+                row = {
+                    'sectionId': item.get('sectionId', ''),
+                    'LoginId': item.get('loginId', ''),
+                    'role': item.get('role', ''),
+                    'ReasonForFailure': item.get('error', '')
+                }
+                writer.writerow(row)
+            csv_content: str = output.getvalue()
+            filename: str = f'course_{course_id}_failures.csv'
+            mime_type: str = 'text/csv'
+            attachment: tuple = (filename, csv_content, mime_type)
+        except (ValueError, Exception) as e:
+            logger.error(f"Failed to create CSV attachment for course {course_id}: {e}")
+
+    logger.info(email_subject)
```

Comment:
This seems like an unnecessary log message comment. I'd either remove it or expand on it to include details like to_email, enrollment_count and failed_enrollments?

Reply:
Ok I expanded on the log statement, `Sending enrollment summary email to janedoe@gmail.edu: For course 403334, 0/7 enrollments finished successfully (7 failed)`